ara/plugins/callback/ara_default.py

```diff
@@ -331,6 +332,10 @@ def v2_playbook_on_start(self, playbook):
         # Lookup the hostname for localhost if necessary
         self.localhost_hostname = self._get_localhost_hostname()
 
+        # TODO: create a function similar to _get_localhost_hostname with error handling (if needed)
+        # Lookup executing username
+        self.executor = pwd.getpwuid(os.getuid())[0]
```

Comment:
out of curiosity, why you decided for this way to get the username. 
Here two function calls are needed, while there are possibilities with just one call, e.g. with the `getpass` library: `getpass.getuser( )`. 
The additional advantage here is also that this function first checks for commen env variables (`USERNAME`, `USER`, `LOGNAME`) and only if they are not set it will use the `pwd` module. 
The pwd module will directly do something similar as `getent passwd`. This is fine on systems with only local user entries. But can be more problematic on systems with e.g. LDAP, because it will increase the LDAP queries (if it is not cached) and will also not work if there any problems with LDAP.

In following example I disabled LDAP to simulate a LDAP error:
```
>>> import getpass
>>> getpass.getuser()
'hille'
>>> import os, pwd
>>> pwd.getpwuid(os.getuid())[0]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 'getpwuid(): uid not found: 28242'
```

While `getpass.getuser()` is still working, `pwd` is not working anymore.

Reply:
Good catch! There's no specific reason *for* my decision, it just was the first (allegedly) reasonably robust implementation that came to mind. This judgement was apparently wrong, I'll go ahead and implement the lookup with `getpass.getuser()`.